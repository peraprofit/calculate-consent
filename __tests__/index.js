import calculateConsent from '../src';
import data from './data.json';

test('calculateConsent default to 15 votes', () => {
  const consent = calculateConsent(data.priorities);
  expect(consent).toMatchSnapshot();
});

test('calculateConsent for 30 votes', () => {
  const consent = calculateConsent(data.priorities, 30);
  expect(consent).toMatchSnapshot();
});
