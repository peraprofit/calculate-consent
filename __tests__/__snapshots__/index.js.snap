// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calculateConsent default to 15 votes 1`] = `
Object {
  "cizx2ezn5ar110160k94gp07k": 1.1866125760649089,
  "cizx2ezntar130160gyt4loff": 1.1866125760649089,
  "cizx2ezocar170160xqt8yah4": 1.2596348884381339,
  "cizx2ezovar190160gchtfdtz": 0.4563894523326572,
  "cizx2ezpdar1b0160sag13b2h": 1.0588235294117647,
  "cizx2ezpwar1e0160qud2624k": 1.2961460446247466,
  "cizx2ezqcar1g01602m49ttx4": 0.6024340770791075,
  "cizx2ezqrar5i0160qkbz11ej": 1.460446247464503,
  "cizx2ezr7ar5k0160iezg83hb": 1.5699797160243407,
  "cizx2ezrmar5m0160x2786i64": 0.6024340770791075,
  "cizx2ezs1ar5o01607snqnjvx": 1.7525354969574036,
  "cizx2ezsjar5q0160lk5zs4ot": 0.5841784989858012,
  "cizx2ezt4ar5t01601wxgo5t2": 1.4056795131845843,
  "cizx2eztjar8z0160qzqcdihe": 0.18255578093306288,
  "cizx2eztzar910160qgsr4uk9": 0.38336713995943206,
  "cizx2ezuear930160bv8yveqb": 1.8255578093306288,
  "cizx2ezusar950160cj5ccd55": 0.127789046653144,
  "cizx2ezv8ar970160t5g2zhsp": 1.0588235294117647,
  "cj10t9xmydmso0145n3ncnndy": 0.14293287848054584,
  "cj10tz1cnb7wv0112stisiqgk": 0.9492900608519271,
  "cj10vao7mxrlu0112ybvu70ai": 0.28113590263691685,
  "cj110z4zpmxrj0145kbqhlnm9": 0.1204868154158215,
  "cj111wgi44lpa01455py4kg8o": 1.1683569979716024,
  "cj126dmu81agq0112jk60b3zd": 0.09438963673243593,
}
`;

exports[`calculateConsent for 30 votes 1`] = `
Object {
  "cizx2ezn5ar110160k94gp07k": 1.977687626774848,
  "cizx2ezntar130160gyt4loff": 1.977687626774848,
  "cizx2ezocar170160xqt8yah4": 2.0993914807302234,
  "cizx2ezovar190160gchtfdtz": 0.7606490872210954,
  "cizx2ezpdar1b0160sag13b2h": 1.7647058823529411,
  "cizx2ezpwar1e0160qud2624k": 2.160243407707911,
  "cizx2ezqcar1g01602m49ttx4": 1.0040567951318458,
  "cizx2ezqrar5i0160qkbz11ej": 2.434077079107505,
  "cizx2ezr7ar5k0160iezg83hb": 2.6166328600405677,
  "cizx2ezrmar5m0160x2786i64": 1.0040567951318458,
  "cizx2ezs1ar5o01607snqnjvx": 2.920892494929006,
  "cizx2ezsjar5q0160lk5zs4ot": 0.973630831643002,
  "cizx2ezt4ar5t01601wxgo5t2": 2.342799188640974,
  "cizx2eztjar8z0160qzqcdihe": 0.3042596348884381,
  "cizx2eztzar910160qgsr4uk9": 0.6389452332657201,
  "cizx2ezuear930160bv8yveqb": 3.0425963488843815,
  "cizx2ezusar950160cj5ccd55": 0.2129817444219067,
  "cizx2ezv8ar970160t5g2zhsp": 1.7647058823529411,
  "cj10t9xmydmso0145n3ncnndy": 0.23822146413424305,
  "cj10tz1cnb7wv0112stisiqgk": 1.5821501014198784,
  "cj10vao7mxrlu0112ybvu70ai": 0.46855983772819476,
  "cj110z4zpmxrj0145kbqhlnm9": 0.20081135902636915,
  "cj111wgi44lpa01455py4kg8o": 1.947261663286004,
  "cj126dmu81agq0112jk60b3zd": 0.15731606122072656,
}
`;
